### ISSUE TITLE
XSS in the folder names

### ISSUE THREAD

**Issue by joanbono** [vakata/jstree#1755]
Hi @vakata , I've discovered a XSS in the folder names, both in JSON and HTML array, when creating the folder tree.

## JSON

~~~~json
$('#container').jstree({
    'core' : {
        'data' : [
            { "text" : "Root node", "children" : [
                { "text" : prompt(/XSS/) },
                { "text" : "Child node 2" }
            ]
            },
        ]
    }
});
~~~~

## HTML

~~~~html
<div id="container">
  <ul>
    <li>Root node
      <ul>
        <li><script>alert("XSS");</script></li>
        <li>Child node 2</li>
      </ul>
    </li>
  </ul>
</div>
~~~~

I don't know if you are going to fix it or if this is a "problem" which the final user has to solve before use the `jstree`, so I decided to let you know the problem.

**Comment by vakata** [comment]
Hi and thank you for the report.

**TLDR:** to prevent executing script tags in the names of loaded nodes you can [use the `force_text` option](https://www.jstree.com/api/#/?q=for&f=$.jstree.defaults.core.force_text)

**Now for the detailed explanation:**
jsTree supports HTML node titles by default - which means adding javascript to the node titles is possible. XSS would imply using user input for creating the tree, which is of course possible and depends on the app logic. 
It is however important to note that jstree will try to prevent script injection when renaming, creating or dragging nodes, so even with HTML node titles enabled this attack vector should be limited to nodes that are loaded (not created manually or renamed). 
If you explicitly provide script tags in the node names when loading the tree - jstree will not filter them out (unless force_text is enabled). 

If you have any ideas on improving the current situation - please share - I am open to suggestions.

Best regards,
Ivan
